src/local_cache.cc

```diff
@@ -271,30 +315,62 @@ LocalCache::Insert(const std::string& key, CacheEntry& entry)
     }
   }
 
-  // Exit early if key already exists to avoid setting up entry unnecessarily
+  return nullptr;  // success
+}
+
+TRITONSERVER_Error*
+LocalCache::Insert(
+    const std::string& key, TRITONCACHE_CacheEntry* entry,
+    TRITONCACHE_Allocator* allocator)
```

Comment:
@GuanLuo from our discussion: On insertion, cache implementation can create/hold it's own TRITONCACHE_CacheEntry, and on Lookup, no need to setup a new one via AddBuffer() calls, you can just return the one being held.

Reply:
Regarding this comment:

CacheEntryNew/Delete APIs were actually never exposed since Triton could create them internally for both Insert/Lookup and it simplified the lifetimes/ownerships of things - so we'd need to expose these APIs for the cache to create/hold/delete its own TRITONCACHE_CacheEntry.

I prefer it this way having less C APIs and simpler lifetimes. I'm not sure how much value having the cache store a TRITONCACHE_CacheEntry object will help or speedup future Lookups as its not very expensive to make the AddBuffer calls, and the performance of this branch is already very close to the old cache from some benchmarking I've been doing. 

Let me know if you want to discuss this more.

Reply:
From Guan: "ship it"